### DISCUSSION TITLE
все работало, решил увеличить матрицу и все сломал

### DISCUSSION THREAD

**Discussion by LAS-Soft** [vvip-68/GyverPanelWiFi#294] [Задавайте вопросы]
доброго всем дня.
собрал матрицу из светодиодной ленты/гирлянды с прозрачным проводом и залитыми диодами 12х12 (144) все прекрасно работало, все эффекты, все подключения. Решил сделать больше и купил еще ленту.
Сегодня она пришла, подключил к 144 еще 200 диодов, и на полу они все засветились, эффекты есть, все нормально работает. Укоротил ленту до 12х12 (144 диодов) Разместил на окне и соединил со старой и всё... больше не работает! горит только один диод зеленым возле точки подачи питания (оно у меня подается в середине а блок управления на одном из концов гирлянды). отключил новую гирлянду, поставил блок как и было - результат тот же! прошил блок еще раз, без успешно.
подскажите что может означать этот зеленый диод? питание на блоке управления проверил - 4,6В.
БП стоит в пластиковом корпусе, на 3А

**Comment by vvip-68** [comment]
Ну если при тесте горело как нужно, а после монтажа - нет - объяснение одно. Где-то оборвались "сопли". Зеленый горящий светодиод в точке питания не означает ничего. Может быть вы при монтаже случайно подали бОльшее напряжение и сгорели либо диоды, либо контроллер?

**Reply by LAS-Soft** [reply]
монтаж вел без питания, контроллер к ПК подключается, прошивается, более того я вполне могу к нему подключится с телефона и управлять им, но вот отображения нет. Может ли быть такое что пин данных переназначился в процессе подключения/отключения контроллера? напряжения больше 5в 3а у меня просто нет в этой системе

**Comment by LAS-Soft** [comment]
сейчас изменил настройку на 12х12, вкл/выкл питание и внезапно матрица ожила. провода не трогал, блок управления тоже:( поработала неменого и на каком-то эффекте зависла не отзывается на команды с приложения - блок управления получается сбойный?

**Reply by vvip-68** [reply]
Он сбойный может быть из за некачественного питания (например шумящий китайский блок питания) или... Даже не знаю еще из за чего. Конкретно в этой версии прошивки с зависаниями не сталкивался при наличии качественного питания

**Reply by LAS-Soft** [reply]
обратил внимание что если из приложения включаю гирлянду то этот диод начинает мигать, если выключаю то он остается светится тем светом которым горел. БП китайский, проверить на шум нечем:( попробую поискать ему замену. Если на контроллер подавать отдельно питание через USB а на ленту отдельно, будет система работать?

**Reply by vvip-68** [reply]
Главное, чтобы минус (земля) были соединены. А так - да, можно и раздельно питать.
Про горящий случайным образом первый диод - обычно когда вместо рекомендованной версии ядра 2,7,4 ставят произвольную первую попавшуюся.

**Reply by LAS-Soft** [reply]
все по рецепту делал
![Снимок экрана 2023-12-21 173123](https://github.com/vvip-68/GyverPanelWiFi/assets/152433165/fc75983e-553c-47e8-bf57-a7481b77a58b)

**Reply by vvip-68** [reply]
Ну да, всё верно

**Reply by LAS-Soft** [reply]
подключил остаток гирлянды (не собирал его в матрицу) из 99 диодов и он нормально работает. но при этом я и блок перемещал и питание отключал полностью:( буду искать сопли. Есть ли способ проверить исправность адресной ленты в идеале без блока управления?

**Reply by vvip-68** [reply]
Нет. Но можно взять любой микроконтроллер доступный, загрузить в него скетч из примеров к FastLED (Файл -> Примеры -> FastLED), исправив там пин подключения и количество диодов.

**Comment by LAS-Soft** [comment]
страннейший глюк который пока не удается локализовать.
подключаю контроллер к новой гирлянде из 99 диодов - в  настройках матрица 12х12 - все 99 диодов работают, режимы меняются, все ОК
подключаю тот же контроллер с тем же БП но к гирлянде в 144 диода что висит на окне и получаю полную недееспособность, лишь иногда что-то как-то проскакивает. контакты вроде все исправны, питание 5в и на блоке и на другом конце гирлянды есть - нарушение в сигнальном контакте?
прошу прощения что столько вопросов но опыта у меня крайне мало в этой сфере, процессы управления/работы не знаю и соответсвенно куда копать тоже не понимаю.
Если все же был баг по питанию, то каким образом выходит из строя гирлянда? не работают диоды вообще или только те что могли сгореть? если при определенном стечении обстоятельств мне удалось сегодня включить все диоды на гирлянде но они не управлялись?

**Reply by vvip-68** [reply]
Вполне мог сгореть какой-то диод (первый, например) и все что дальше него - сигнал не передается, "изображения" нет.
Есть возможность разъединить куски и подключить для проверки линию данных к другому куску. Или отрезать первый диод, подключиться ко второму. Если засветятся - значит первый неисправен и его нужно менять...

**Reply by LAS-Soft** [reply]
они при каких-то манипуляциях с блоком (я бы сказал каким-то чудом) периодически светятся все, но не в режиме и не управляются. бывает такое что светятся и управляются первые три. Гирлянду на сегменты разделял и блок управления по ней перемещал но результата не достиг. Исходя из того что отдельная гирлянда работает корректно даже с этим БП делаю вывод что контроллер исправен. Нарезать всю гирлянду на отдельные диоды не улыбается:( они залитые, была бы пайка не было бы вопросов. Собственно поэтому и спрашиваю как они выходят из строя и как это проверить

**Reply by vvip-68** [reply]
Как выходят из строя? При скачек напряжения питания диодов выше допустимого микросхема внутри диода сгорает. Может один, которому не повезло, может несколько. А могут и сами по себе сгореть, но я правда на практике с таким не сталкивался. Поскольку диод "умный" и реагирует только на определенный сигнал на входе - то проверить его можно исключительно подключив к микроконтроллеру, который будет формировать эту управляющую последовательность. Другого варианта нет.

Обычно неисправные диоды (по крайней мере на плоской матрице) могут нагреваться. Но без тепловизора вы это вряд ли увидите... У вас есть тепловизор? Тогда смотрите какие диоды горячее остальных.

**Reply by LAS-Soft** [reply]
Если человек идиот, это навсегда! я при наращивании перепутал направление диодов! сейчас перепаяю все и буду пробовать еще раз. Спасибо за помощь

**Comment by LAS-Soft** [comment]
все заработало! Еще раз спасибо, наставили на пусть истинный.
Собственно что произошло:
первый раз я купил 20м гирлянды (200 диодов) для опытов и понять насколько вообще ярко/красиво планировал по 50 диодов на одно стекло. Собрал, повесил и понял что очень жидко выглядит, сделал по 72 диода на стекло - стало значительно лучше, ну я и заказал еще 20м. 
Так как от первой гирлянды оставался хвост в 56 диодов я решил его использовать и припаял новую к нему, вот тут я и совершил ошибку! я спаял верно по полярности, но в обратную сторону по сигналу! получилось 56 в правильную сторону и 88 в обратную! соответственно куда бы в этой консерватории я не внедрял контроллер он получается не мог сформировать корректный сигнал.
Остался у меня только вопрос почему он не работал когда я подключал его к нормально собранной старой гирлянде, но да это уже и не особо важно, главное что теперь все работает

**Reply by vvip-68** [reply]
И это вам еще повезло, что докупленная гирлянда имела один и тот же порядок цветов в диодах. Я вот заказывал через время у одного и того же продавца - 7 гирлянд по 5 метров, позже  кусок 15 метров плюс еще 5 метров. Первые 7 кусков (возьмем за образец) светятся красным, тот что 15 метров - зеленым, а тот что 5 - вообще синим. И вот что мне теперь с этим всем делать? :(
У всех внутри диодов оказался разный порядок цветов...

**Reply by LAS-Soft** [reply]
мне показался немного иной оттенок, но монотонными цветами пока не заливал да и у меня два одинаковых окна рядом, так что смотрится все равно гармонично. Вообще посетила мысль сделать на след год шаг 5см между диодами но получается море пайки и работы с проводами, как альтернативу рассматриваю ленту с шагом 3см, разрешение будет много лучше и эффекты можно прокручивать круче (вплоть до взаимодействия двух окон типа игры в самый первый теннис:))

**Reply by vvip-68** [reply]
Ну - "Время - деньги". Либо деньги либо время. На али есть такие же прозрачно-проводные нити с диодами, залитыми в матово-белые капельки и с шагом 5 см... Есть спрос - есть предложение. Китайцы работают.

Навскидку - вот: https://aliexpress.ru/item/1005005656803525.html
Тут 1.5, 2, 2.5, 3 см - на любой вкус. Правда с контроллером. Но попадались лоты - любой длины просто на бобину намотанные - хоть 100 метров.
Вот тут - 5см в бобине по 1000 штук - https://aliexpress.ru/item/1005003829469536.html
Или такая: https://aliexpress.ru/item/32617501443.html
Ну и так далее

**Reply by LAS-Soft** [reply]
да, цветопередача у новой другая:(
Так же осознал почему перевернул направление гирлянды: я наращивал со стороны контроллера и мне нужно было отрезать от конца а не от начала новой (надеюсь понятно описал).
За ссылку на 5см варианты огромное спасибо, мне почему-то не удавалось найти такие
